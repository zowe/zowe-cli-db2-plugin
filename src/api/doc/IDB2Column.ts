/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

/**
 * Column description
 * @interface IDB2Column
 * @export
 */
export type IDB2Column = {
    /**
     * The catalog name
     * @type {string}
     * @memberof IDB2Column
     */
    TABLE_CAT: string;

    /**
     * The scheme name
     * @type {string}
     * @memberof IDB2Column
     */
    TABLE_SCHEM: string;

    /**
     * The table name
     * @type {string}
     * @memberof IDB2Column
     */
    TABLE_NAME: string;

    /**
     * The column name
     * @type {string}
     * @memberof IDB2Column
     */
    COLUMN_NAME: string;

    /**
     * The numeric representation of the SQL data type
     * @type {number}
     * @memberof IDB2Column
     */
    DATA_TYPE: number;

    /**
     * The string representation of the SQL data type
     * @type {string}
     * @memberof IDB2Column
     */
    TYPE_NAME: string;

    /**
     * The length of the column
     * @type {number}
     * @memberof IDB2Column
     */
    COLUMN_SIZE: number;

    /**
     * Buffer length needed to store data
     * @type {number}
     * @memberof IDB2Column
     */
    BUFFER_LENGTH: number;

    /**
     * Number of decimal digits for a numeric column
     * @type {number}
     * @memberof IDB2Column
     */
    DECIMAL_DIGITS: number | null;

    /**
     * The radix of a numeric column
     * @type {number}
     * @memberof IDB2Column
     */
    NUM_PREC_RADIX: number | null;

    /**
     * Indicates whether the column can contain the null value
     *
     * 0 - NULLs are not allowed
     * 1 - NULLs are allowed
     * @type {number}
     * @memberof IDB2Column
     */
    NULLABLE: number;

    /**
     * Description of the column
     * @type {string}
     * @memberof IDB2Column
     */
    REMARKS: string;

    /**
     * The default value
     * @type {string}
     * @memberof IDB2Column
     */
    COLUMN_DEF: string;

    /**
     * The numeric representation of the SQL data type
     * @type {number}
     * @memberof IDB2Column
     */
    SQL_DATA_TYPE: number;

    /**
     * The subtype of a datetime type
     *
     * 1 - Date
     * 2 - Time
     * 3 - Timestamp
     * @type {number|null}
     * @memberof IDB2Column
     */
    SQL_DATETIME_SUB: number | null;

    /**
     * Length in bytes of the column
     * @type {number|null}
     * @memberof IDB2Column
     */
    CHAR_OCTET_LENGTH: number | null;

    /**
     * Position of the column in the table starting from 1
     * @type {number}
     * @memberof IDB2Column
     */
    ORDINAL_POSITION: number;

    /**
     * Indicates whether the column can contain the null value
     *
     * NO - NULLs are not allowed
     * YES - NULLs are allowed
     * @type {string}
     * @memberof IDB2Column
     */
    IS_NULLABLE: string;
};
