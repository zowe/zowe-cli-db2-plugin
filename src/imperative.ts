/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { IImperativeConfig } from "@brightside/imperative";
import { Constants } from "./";

const config: IImperativeConfig = {
    name: Constants.PROD_NAME,
    commandModuleGlobs: ["**/cli/*/*.definition!(.d).*s"],
    rootCommandDescription: Constants.DESCRIPTION,
    productDisplayName: Constants.DISPLAY_NAME,
    pluginHealthCheck: "./lib/HealthCheck.handler",
    profiles: [
        {
            type: "db2",
            schema: {
                type: "object",
                title: "Db2 Profile",
                description: "A profile for interaction with Db2 for the z/OS region",
                properties: {
                    host: {
                        type: "string",
                        optionDefinition: {
                            name: "host",
                            aliases: ["H"],
                            description: "The Db2 server host name",
                            type: "string",
                        },
                    },
                    port: {
                        type: "number",
                        optionDefinition: {
                            name: "port",
                            aliases: ["P"],
                            description: "The Db2 server port number",
                            type: "number",
                        },
                    },
                    user: {
                        type: "string",
                        secure: true,
                        optionDefinition: {
                            name: "user",
                            aliases: ["u"],
                            description: "The Db2 user ID (may be the same as the TSO login)",
                            type: "string",
                        },
                    },
                    password: {
                        type: "string",
                        secure: true,
                        optionDefinition: {
                            name: "password",
                            aliases: ["pass", "pw"],
                            description: "The Db2 password (may be the same as the TSO password)",
                            type: "string",
                        },
                    },
                    database: {
                        type: "string",
                        optionDefinition: {
                            name: "database",
                            aliases: ["d"],
                            description: "The name of the database",
                            type: "string",
                        },
                    },
                    sslFile: {
                        type: "string",
                        optionDefinition: {
                            name: "ssl-file",
                            aliases: ["s"],
                            description: "Path to an SSL Certificate file",
                            type: "string",
                        },
                    },
                },
            },
        }
    ],
};

export = config;
