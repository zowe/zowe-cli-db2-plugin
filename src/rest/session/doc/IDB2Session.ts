/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

/**
 * Interface for DB2 connection parameters
 * @export
 * @interface
 */
export interface IDB2Session {

    /**
     * The name or IP address of the server to connect to
     * @type {string}
     * @memberof IDB2Session
     */
    hostname: string;

    /**
     * The port number where the server is listening
     * @type {number}
     * @memberof IDB2Session
     */
    port: number;

    /**
     * The user ID
     * @type {string}
     * @memberof IDB2Session
     */
    user: string;

    /**
     * The user's password
     * @type {string}
     * @memberof IDB2Session
     */
    password: string;

    /**
     * The database name to use
     * @type {string}
     * @memberof IDB2Session
     */
    database: string;

    /**
     * The path to a SSL Certificate file from a server or a CA signed certificate
     *
     * When specified the connection to the database server will be done over SSL.
     * @type {string}
     * @memberof IDB2Session
     */
    sslFile?: string;
}
