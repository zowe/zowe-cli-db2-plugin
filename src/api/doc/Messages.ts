/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { IMessageDefinition, apiErrorHeader } from "@zowe/imperative";

export const noDB2Input: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 parameters were supplied.",
};

export const noHostName: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 server hostname was supplied.",
};

export const noPortNumber: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 server port number was supplied.",
};

export const noUserName: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 User ID was supplied.",
};

export const noPassword: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 user password was supplied.",
};

export const noDatabaseName: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 database name was supplied.",
};

export const noTableName: IMessageDefinition = {
    message: apiErrorHeader.message + " - No DB2 table name was supplied.",
};
