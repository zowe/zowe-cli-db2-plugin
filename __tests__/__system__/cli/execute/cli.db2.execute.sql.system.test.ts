/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { ITestEnvironment, TestEnvironment, runCliScript } from "@zowe/cli-test-utils";
import { ITestPropertiesSchema } from "../../../__src__/environment/doc/ITestPropertiesSchema";
import { ICommandResponse } from "@zowe/imperative";

let TEST_ENV: ITestEnvironment<ITestPropertiesSchema>;
let host: string;
let port: number;
let user: string;
let password: string;
let database: string;
describe("db2 execute sql command", () => {

    // Create the unique test environment
    beforeAll(async () => {
        TEST_ENV = await TestEnvironment.setUp({
            installPlugin: true,
            tempProfileTypes: ["db2"],
            testName: "execute_sql_command",
        });

        host = TEST_ENV.systemTestProperties.db2.host;
        port = TEST_ENV.systemTestProperties.db2.port;
        user = TEST_ENV.systemTestProperties.db2.user;
        password = TEST_ENV.systemTestProperties.db2.password;
        database = TEST_ENV.systemTestProperties.db2.database;
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENV);
    });

    it("should display the help", () => {
        const response = runCliScript(__dirname + "/__scripts__/sql_help.sh", TEST_ENV);
        expect(response.stdout.toString()).toMatchSnapshot();
        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
    });

    it("should fail with invalid option", () => {
        const response = runCliScript(__dirname + "/__scripts__/sql_invalid_option.sh", TEST_ENV);
        expect(response.stderr.toString()).toMatchSnapshot();
        expect(response.stdout.toString()).toMatchSnapshot();
        expect(response.status).toBe(1);
    });

    it("should not accept mutually exclusive options", () => {
        const response = runCliScript(__dirname + "/__scripts__/fail_query_file.sh", TEST_ENV);
        expect(response.stderr.toString()).toMatchSnapshot();
        expect(response.stdout.toString()).toMatchSnapshot();
        expect(response.status).toBe(1);
    });

    it("should complete the command successfully and return a valid JSON response", () => {
        const response = runCliScript(__dirname + "/__scripts__/success_rfj.sh", TEST_ENV);
        expect(response.stderr.toString()).toBe("");
        expect(response.stdout.toString()).toMatchSnapshot();
        expect(response.status).toBe(0);
        // Convert response to an object and check fields
        const respObj: ICommandResponse  = JSON.parse(response.stdout.toString());
        expect(respObj.success).toBe(true);
        expect(respObj.message).toBe("");
        expect(respObj.stderr).toBe("");
        expect(respObj.error).toBeUndefined();
        expect((respObj.data as any[]).length).toBeGreaterThan(0);
    });

    it("should be able to execute SQL statements from file", () => {
        const response = runCliScript(__dirname + "/__scripts__/success_file.sh", TEST_ENV);
        expect(response.stderr.toString()).toBe("");
        expect(response.stdout.toString()).toMatchSnapshot();
        expect(response.status).toBe(0);
    });


    it("should be able to execute SQL statements overriding some of the options by arguments", () => {
        const response = runCliScript(__dirname + "/__scripts__/success_override_profile.sh",
            TEST_ENV, [host, port, database]);
        expect(response.stderr.toString()).toBe("");
        expect(response.stdout.toString()).toMatchSnapshot();
        expect(response.status).toBe(0);
    });

    it("should fail when file doesn't exist", () => {
        const response = runCliScript(__dirname + "/__scripts__/fail_no_file.sh", TEST_ENV);
        expect(response.stderr.toString()).toMatchSnapshot();
        expect(response.stdout.toString()).toBe("");
        expect(response.status).toBe(1);
    });
});

describe("should execute sql commands without profile", () => {
    // Create the unique test environment
    beforeAll(async () => {
        TEST_ENV = await TestEnvironment.setUp({
            installPlugin: true,
            testName: "execute_sql_command",
        });

        host = TEST_ENV.systemTestProperties.db2.host;
        port = TEST_ENV.systemTestProperties.db2.port;
        user = TEST_ENV.systemTestProperties.db2.user;
        password = TEST_ENV.systemTestProperties.db2.password;
        database = TEST_ENV.systemTestProperties.db2.database;
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENV);
    });

    it("should be able to execute SQL statements using passed options not profile", () => {
        const response = runCliScript(__dirname + "/__scripts__/success_no_profile.sh",
            TEST_ENV, [host, port, user, password, database]);
        expect(response.stderr.toString()).toBe("");
        expect(response.stdout.toString()).toMatchSnapshot();
        expect(response.status).toBe(0);
    });
});
