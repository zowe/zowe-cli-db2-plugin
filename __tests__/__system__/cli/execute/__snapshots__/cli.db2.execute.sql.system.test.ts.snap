// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`db2 execute sql command should be able to execute SQL statements from file 1`] = `
"Result #1
- 
  TOTAL: 1


Result #2
- 
  DATA: TEXT


"
`;

exports[`db2 execute sql command should be able to execute SQL statements overriding some of the options by arguments 1`] = `
"Result #1
- 
  TOTAL: 1


"
`;

exports[`db2 execute sql command should complete the command successfully and return a valid JSON response 1`] = `
"{
  \\"success\\": true,
  \\"exitCode\\": 0,
  \\"message\\": \\"\\",
  \\"stdout\\": \\"Result #1\\\\n- \\\\n  TOTAL: 1\\\\n\\\\n\\\\n\\",
  \\"stderr\\": \\"\\",
  \\"data\\": [
    [
      {
        \\"TOTAL\\": 1
      }
    ]
  ]
}"
`;

exports[`db2 execute sql command should display the help 1`] = `
"
 COMMAND NAME
 ------------

   sql

 DESCRIPTION
 -----------

   Execute one or multiple SQL statements separated by a semicolon from a command
   line or from a file.

 USAGE
 -----

   zowe db2 execute sql [options]

 OPTIONS
 -------

   --query  | -q (string)

      The SQL statement verbatim to execute

   --file  | -f (string)

      A local file containing the SQL statements to execute

 DB2 CONNECTION OPTIONS
 ----------------------

   --host  | -H (string)

      The Db2 server host name

   --port  | -P (number)

      The Db2 server port number

   --user  | -u (string)

      The Db2 user ID (may be the same as the TSO login)

   --password  | --pass | --pw (string)

      The Db2 password (may be the same as the TSO password)

   --database  | --db (string)

      The name of the database

   --sslFile  | --ssl (string)

      Path to an SSL Certificate file

 PROFILE OPTIONS
 ---------------

   --db2-profile  | --db2-p (string)

      The name of a (db2) profile to load for this command execution.

   --base-profile  | --base-p (string)

      The name of a (base) profile to load for this command execution.

 BASE CONNECTION OPTIONS
 -----------------------

   --reject-unauthorized  | --ru (boolean)

      Reject self-signed certificates.

      Default value: true

   --token-type  | --tt (string)

      The type of token to get and use for the API. Omit this option to use the
      default token type, which is provided by 'zowe auth login'.

   --token-value  | --tv (string)

      The value of the token to pass to the API.

 GLOBAL OPTIONS
 --------------

   --response-format-json  | --rfj (boolean)

      Produce JSON formatted data from a command

   --help  | -h (boolean)

      Display help text

   --help-examples  (boolean)

      Display examples for all the commands in a group

   --help-web  | --hw (boolean)

      Display HTML help in browser

 EXAMPLES
 --------

   - Execute a dummy SQL query:

      $ zowe db2 execute sql --query \\"SELECT 'Hello World' FROM SYSIBM.SYSDUMMY1\\"

   - Retrieve the employees table and total number of rows:

      $ zowe db2 execute sql -q \\"SELECT * FROM SAMPLE.EMP; SELECT COUNT(*) AS TOTAL FROM SAMPLE.EMP\\"

   - Execute a file with SQL statements:

      $ zowe db2 execute sql --file backup_sample_database.sql

{
  \\"success\\": true,
  \\"exitCode\\": 0,
  \\"message\\": \\"The help was constructed for command: sql.\\",
  \\"stdout\\": \\"\\\\n COMMAND NAME\\\\n ------------\\\\n\\\\n   sql\\\\n\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Execute one or multiple SQL statements separated by a semicolon from a command\\\\n   line or from a file.\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   zowe db2 execute sql [options]\\\\n\\\\n OPTIONS\\\\n -------\\\\n\\\\n   --query  | -q (string)\\\\n\\\\n      The SQL statement verbatim to execute\\\\n\\\\n   --file  | -f (string)\\\\n\\\\n      A local file containing the SQL statements to execute\\\\n\\\\n DB2 CONNECTION OPTIONS\\\\n ----------------------\\\\n\\\\n   --host  | -H (string)\\\\n\\\\n      The Db2 server host name\\\\n\\\\n   --port  | -P (number)\\\\n\\\\n      The Db2 server port number\\\\n\\\\n   --user  | -u (string)\\\\n\\\\n      The Db2 user ID (may be the same as the TSO login)\\\\n\\\\n   --password  | --pass | --pw (string)\\\\n\\\\n      The Db2 password (may be the same as the TSO password)\\\\n\\\\n   --database  | --db (string)\\\\n\\\\n      The name of the database\\\\n\\\\n   --sslFile  | --ssl (string)\\\\n\\\\n      Path to an SSL Certificate file\\\\n\\\\n PROFILE OPTIONS\\\\n ---------------\\\\n\\\\n   --db2-profile  | --db2-p (string)\\\\n\\\\n      The name of a (db2) profile to load for this command execution.\\\\n\\\\n   --base-profile  | --base-p (string)\\\\n\\\\n      The name of a (base) profile to load for this command execution.\\\\n\\\\n BASE CONNECTION OPTIONS\\\\n -----------------------\\\\n\\\\n   --reject-unauthorized  | --ru (boolean)\\\\n\\\\n      Reject self-signed certificates.\\\\n\\\\n      Default value: true\\\\n\\\\n   --token-type  | --tt (string)\\\\n\\\\n      The type of token to get and use for the API. Omit this option to use the\\\\n      default token type, which is provided by 'zowe auth login'.\\\\n\\\\n   --token-value  | --tv (string)\\\\n\\\\n      The value of the token to pass to the API.\\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (boolean)\\\\n\\\\n      Produce JSON formatted data from a command\\\\n\\\\n   --help  | -h (boolean)\\\\n\\\\n      Display help text\\\\n\\\\n   --help-examples  (boolean)\\\\n\\\\n      Display examples for all the commands in a group\\\\n\\\\n   --help-web  | --hw (boolean)\\\\n\\\\n      Display HTML help in browser\\\\n\\\\n EXAMPLES\\\\n --------\\\\n\\\\n   - Execute a dummy SQL query:\\\\n\\\\n      $ zowe db2 execute sql --query \\\\\\"SELECT 'Hello World' FROM SYSIBM.SYSDUMMY1\\\\\\"\\\\n\\\\n   - Retrieve the employees table and total number of rows:\\\\n\\\\n      $ zowe db2 execute sql -q \\\\\\"SELECT * FROM SAMPLE.EMP; SELECT COUNT(*) AS TOTAL FROM SAMPLE.EMP\\\\\\"\\\\n\\\\n   - Execute a file with SQL statements:\\\\n\\\\n      $ zowe db2 execute sql --file backup_sample_database.sql\\\\n\\\\n\\",
  \\"stderr\\": \\"\\",
  \\"data\\": \\"\\\\n COMMAND NAME\\\\n ------------\\\\n\\\\n   sql\\\\n\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Execute one or multiple SQL statements separated by a semicolon from a command\\\\n   line or from a file.\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   zowe db2 execute sql [options]\\\\n\\\\n OPTIONS\\\\n -------\\\\n\\\\n   --query  | -q (string)\\\\n\\\\n      The SQL statement verbatim to execute\\\\n\\\\n   --file  | -f (string)\\\\n\\\\n      A local file containing the SQL statements to execute\\\\n\\\\n DB2 CONNECTION OPTIONS\\\\n ----------------------\\\\n\\\\n   --host  | -H (string)\\\\n\\\\n      The Db2 server host name\\\\n\\\\n   --port  | -P (number)\\\\n\\\\n      The Db2 server port number\\\\n\\\\n   --user  | -u (string)\\\\n\\\\n      The Db2 user ID (may be the same as the TSO login)\\\\n\\\\n   --password  | --pass | --pw (string)\\\\n\\\\n      The Db2 password (may be the same as the TSO password)\\\\n\\\\n   --database  | --db (string)\\\\n\\\\n      The name of the database\\\\n\\\\n   --sslFile  | --ssl (string)\\\\n\\\\n      Path to an SSL Certificate file\\\\n\\\\n PROFILE OPTIONS\\\\n ---------------\\\\n\\\\n   --db2-profile  | --db2-p (string)\\\\n\\\\n      The name of a (db2) profile to load for this command execution.\\\\n\\\\n   --base-profile  | --base-p (string)\\\\n\\\\n      The name of a (base) profile to load for this command execution.\\\\n\\\\n BASE CONNECTION OPTIONS\\\\n -----------------------\\\\n\\\\n   --reject-unauthorized  | --ru (boolean)\\\\n\\\\n      Reject self-signed certificates.\\\\n\\\\n      Default value: true\\\\n\\\\n   --token-type  | --tt (string)\\\\n\\\\n      The type of token to get and use for the API. Omit this option to use the\\\\n      default token type, which is provided by 'zowe auth login'.\\\\n\\\\n   --token-value  | --tv (string)\\\\n\\\\n      The value of the token to pass to the API.\\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (boolean)\\\\n\\\\n      Produce JSON formatted data from a command\\\\n\\\\n   --help  | -h (boolean)\\\\n\\\\n      Display help text\\\\n\\\\n   --help-examples  (boolean)\\\\n\\\\n      Display examples for all the commands in a group\\\\n\\\\n   --help-web  | --hw (boolean)\\\\n\\\\n      Display HTML help in browser\\\\n\\\\n EXAMPLES\\\\n --------\\\\n\\\\n   - Execute a dummy SQL query:\\\\n\\\\n      $ zowe db2 execute sql --query \\\\\\"SELECT 'Hello World' FROM SYSIBM.SYSDUMMY1\\\\\\"\\\\n\\\\n   - Retrieve the employees table and total number of rows:\\\\n\\\\n      $ zowe db2 execute sql -q \\\\\\"SELECT * FROM SAMPLE.EMP; SELECT COUNT(*) AS TOTAL FROM SAMPLE.EMP\\\\\\"\\\\n\\\\n   - Execute a file with SQL statements:\\\\n\\\\n      $ zowe db2 execute sql --file backup_sample_database.sql\\\\n\\\\n\\"
}"
`;

exports[`db2 execute sql command should fail when file doesn't exist 1`] = `
"Command Error:
Error: ENOENT: no such file or directory, open 'not-existing-file.sql'
"
`;

exports[`db2 execute sql command should fail with invalid option 1`] = `
"Command Error:
Unknown argument: commit
Command failed due to improper syntax
Command entered: \\"zowe db2 execute sql --commit SELECT 1 FROM SYSIBM.SYSDUMMY1\\"
Available commands are \\"sql\\".
Use \\"zowe db2 execute --help\\" to view groups, commands, and options.
Error Details:
Unknown argument: commit
"
`;

exports[`db2 execute sql command should fail with invalid option 2`] = `""`;

exports[`db2 execute sql command should not accept mutually exclusive options 1`] = `
"
Syntax Error:
The following options conflict (mutually exclusive):
--query
--file

Syntax Error:
The following options conflict (mutually exclusive):
--file
--query

Example:

 - Execute a dummy SQL query:

      $ zowe db2 execute sql --query \\"SELECT 'Hello World' FROM SYSIBM.SYSDUMMY1\\"

Use \\"zowe db2 execute sql --help\\" to view command description, usage, and options.
"
`;

exports[`db2 execute sql command should not accept mutually exclusive options 2`] = `""`;

exports[`should execute sql commands without profile should be able to execute SQL statements using passed options not profile 1`] = `
"Result #1
- 
  TOTAL: 1


"
`;
